cloud cli updates (#4354)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/cloud-cli-installation.md

The dbt Cloud CLI is available in all [deployment regions](/docs/cloud/about-cloud/regions-ip-addresses) and for both multi-tenant and single-tenant accounts (Azure single-tenant not supported at this time).

- Ensure you are using dbt version 1.5 or higher. Refer to [dbt Cloud versions](/docs/dbt-versions/upgrade-core-in-cloud) to upgrade.

> - SSH tunneling for [Postgres and Redshift](/docs/cloud/connect-data-platform/connect-redshift-postgresql-alloydb) connections, [PrivateLink](/docs/cloud/secure/about-privatelink), and [Single sign-on (SSO)](/docs/cloud/manage-access/sso-overview) isn't supported yet.

Author: ```suggestion
- Note that SSH tunneling for [Postgres and Redshift](/docs/cloud/connect-data-platform/connect-redshift-postgresql-alloydb) connections, [PrivateLink](/docs/cloud/secure/about-privatelink), and [Single sign-on (SSO)](/docs/cloud/manage-access/sso-overview) isn't supported yet.
```

> - Note that SSH tunneling for [Postgres and Redshift](/docs/cloud/connect-data-platform/connect-redshift-postgresql-alloydb) connections, [PrivateLink](/docs/cloud/secure/about-privatelink), and [Single sign-on (SSO)](/docs/cloud/manage-access/sso-overview) isn't supported yet.

Reviewer: PrivateLink will be supported as of tomorrow!

Author: awesome! will it be compatible with the SL ? cc @aiguofer

> - Note that SSH tunneling for [Postgres and Redshift](/docs/cloud/connect-data-platform/connect-redshift-postgresql-alloydb) connections, [PrivateLink](/docs/cloud/secure/about-privatelink), and [Single sign-on (SSO)](/docs/cloud/manage-access/sso-overview) isn't supported yet.

Author: ```suggestion
- Note that SSH tunneling for [Postgres and Redshift](/docs/cloud/connect-data-platform/connect-redshift-postgresql-alloydb) connections and [Single sign-on (SSO)](/docs/cloud/manage-access/sso-overview) doesn't support the dbt Cloud CLI yet.
```

- Ensure you are using dbt version 1.5 or higher. Refer to [dbt Cloud versions](/docs/dbt-versions/upgrade-core-in-cloud) to upgrade.
- Avoid using SSH tunneling for [Postgres and Redshift](/docs/cloud/connect-data-platform/connect-redshift-postgresql-alloydb) connections.
- Avoid using [PrivateLink](/docs/cloud/secure/about-privatelink).

> - Avoid using Single sign-on (SSO) for Snowflake or SSH tunnels for Redshift.

Author: hey @aiguofer ! i've added this bit here per the slack discussion but wanted to clarify if you meant avoid using sso for snowflake only? or also all other adapters? so:

- Ensure you are using dbt version 1.5 or higher. Refer to [dbt Cloud versions](/docs/dbt-versions/upgrade-core-in-cloud) to upgrade.
- Avoid using SSH tunneling for [Postgres and Redshift](/docs/cloud/connect-data-platform/connect-redshift-postgresql-alloydb) connections.
- Avoid using [PrivateLink](/docs/cloud/secure/about-privatelink).
- Avoid using Single sign-on (SSO) for Snowflake - WHAT ABOUT OTHER ADAPTERS?

Reviewer 2: I believe this limitation would apply to all SSO integrations.

> - Avoid using Single sign-on (SSO) if developing for the [dbt Semantic Layer](/docs/use-dbt-semantic-layer/dbt-sl).

Author: ```suggestion
```

<TabItem value="pip" label="Existing dbt Core users (pip)">

If you already have dbt Core installed, the dbt Cloud CLI may conflict. Here are some considerations:

> - **Prevent conflicts** <br /> If you want to use `pip`, create a new virtual environment.<br /><br />

Reviewer: I think "Use both dbt Cloud CLI and dbt Core with pip" is the better heading here.

> - **Use both dbt Cloud CLI and dbt Core** <br /> If you use Homebrew, consider aliasing the dbt Cloud CLI as "dbt-cloud" to avoid conflict. For more details, check the [FAQs](#faqs) if your operating system experiences path conflicts.<br /><br />

Reviewer: "...with brew or native installs"

> - **Switch from the dbt Cloud CLI to dbt Core** <br />

Reviewer: "Reverting back to dbt Core from the dbt Cloud CLI"

<TabItem value="pip" label="Existing dbt Core users (pip)">

> If you already have dbt Core installed, you can install the dbt Cloud CLI using pip. Here are some considerations:

Reviewer: We say below to not use pip, so I would say "If you already have dbt Core installed, the dbt Cloud CLI may conflict. Here are some considerations:"

- **Prevent conflicts**<br/>

>   To prevent overwriting dbt Core, avoid installing the dbt Cloud CLI with `pip`. Instead, consider using the native installation method (Homebrew) and configure your PATH or create a new virtual environment.<br/>

Reviewer: I would say this bullet should be "If you want to use pip, create a new virtual environment"

- **Use both dbt Cloud CLI and dbt Core**<br/>

>   Have both the dbt Cloud CLI and dbt Core installed simultaneously to meet your needs. To avoid conflicts, alias the dbt Cloud CLI as "dbt-cloud." For more details, check the [FAQs](#faqs) if your operating system experiences path conflicts.<br/>

Reviewer: I think this should be "if you use brew, consider aliasing the cloud cli to avoid conflicts". If you install w pip, it will already have overwritten dbt, so this doesn't work there. I would move this to the brew section.

  pip3 install dbt
  ```

3. (Optional) To revert back to dbt Core, first uninstall both the dbt Cloud CLI and dbt Core. Then reinstall dbt Core using the version from Step 1.

  ```bash
  pip3 uninstall dbt-core dbt

>   pip3 install dbt-core==VERSION

Reviewer: This should be

```pip install dbt-adapter_name --force-reinstall```

</Tabs>

> ## Using VS Code extensions

Author: added your snippet @greg-mckeon

Visual Studio (VS) Code extensions enhance command line tools by adding extra functionalities. The dbt Cloud CLI is fully compatible with dbt Core, however it doesn't support some dbt Core APIs required by certain tools, for example VS Code extensions. 

> To use these extensions, such as dbt-power-user, with the dbt Cloud CLI, you can install it using Homebrew (along with dbt Core) and create an alias to run the dbt Cloud CLI as `dbt-cloud`. This allows dbt-power-user to continue to work alongside the dbt Cloud CLI.

Reviewer: "This allows dbt-power-user to continue to invoke dbt Core under the hood, alongside the dbt Cloud CLI."

</Tabs>

## dbt Cloud CLI extensions
Command line extensions are additional functionalities that you can add to a command line tool or shell environment. These extensions expand the capabilities of the CLI, making it more versatile and efficient.

> The dbt Cloud CLI doesn't currently support extensions, such as SQLFluff or dbt-power-user, during the public preview period. This is because.....

Reviewer: I would rewrite this section as:

## Using VS Code extensions with the Cloud CLI
The Cloud CLI is fully compatible with dbt Core, but does not provide certain Core APIs that some tools, particularly VS Code extensions, need to run. 

If you need to use VS Code extensions like dbt-power-user with the Cloud CLI, install via the brew method alongside a dbt Core installation and alias the Cloud CLI to run as `dbt-cloud`. This will allow dbt-power-user to continue to function alongside the Cloud CLI.

<details>
<summary>How do I run both the dbt Cloud CLI and dbt Core?</summary>
For compatibility, both the dbt Cloud CLI and dbt Core are invoked by running <code>dbt</code>. This can create path conflicts if your operating system selects one over the other based on your $PATH environment variable (settings).<br />

If you have dbt Core installed locally, either:

1. Install using the <code>pip3 install dbt</code> [pip](/docs/cloud/cloud-cli-installation?install=pip#install-dbt-cloud-cli) command, or
2. Install natively, ensuring you either deactivate the virtual environment containing dbt Core or create an alias for the dbt Cloud CLI. 
3. (Advanced users) Install natively, but modify the $PATH environment variable to correctly point to the dbt Cloud CLI binary to use both dbt Cloud CLI and dbt Core together.

You can always uninstall the dbt Cloud CLI to return to using dbt Core.
</details>

<details>

> <summary>How to create an alias?</summary>

Author: added this step @greg-mckeon - can you verify it's right?

To create an alias for the dbt Cloud CLI: <br />

> 1. Open your shell's profile configuration file. Deepending on your shell and system, this could be <code>~/.bashrc</code>, <code>~/.bash_profile</code>, <code>~/.zshrc</code>, or another file.<br />

Reviewer: typo: depending

2. Add an alias that points to the dbt Cloud CLI binary. For example:<code>alias dbt-cloud="path_to_dbt_cloud_cli_binary</code>
   

>    Replace <code>path_to_dbt_cloud_cli_binary</code> with the actual path to the dbt Cloud CLI binary. With this alias, you can use the command <code>dbt-cloud</code> to invoke the dbt Cloud CLI.<br />

Reviewer: We should add the default installation location for brew (/opt/homebrew/bin/dbt).

3. Save the file and then either restart your shell or run <code>source</code> on the profile file to apply the changes.
As an example, in bash you would run: <code>source ~/.bashrc</code><br />

> This alias will allow you to use the <code>dbt-cloud</code> command to invoke the dbt Cloud CLI while having dbt Core installed natively.

Reviewer: I would have a step here where they try to run "dbt-cloud" and see the cloud CLI, and "dbt" and see dbt Core.

Author: thank you, added
![Screenshot 2023-11-08 at 12 26 52](https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/a6714427-a10d-47bb-bf8a-6d2fda9e0b42)
!

If you have dbt Core installed locally, either:

- Install using the <code>pip3 install dbt</code> [pip](/docs/cloud/cloud-cli-installation?install=pip#install-dbt-cloud-cli) command, or

> - Install natively, however ensure you deactivate your Python environment or uninstall it using `pip uninstall dbt` before proceeding.  

Reviewer: "Install natively, ensuring that you either deactivate the virtual environment containing core or create an alias for the cloud cli."

It seems "creating an alias for the cloud cli" needs its own section?
